crates/superposition_types/src/lib.rs

```diff
@@ -74,3 +77,307 @@ impl FromRequest for User {
         }
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Copy, Serialize)]
+pub struct Cac<T>(T);
```

Comment:
Can we implement deref trait instead of `into_inner`?

Reply:
@Datron into_inner gives the type wrapped inside i.e T, 
Deref trait mainly dereference the the value , 
and also Deref wont work in this case as Deref on CAC requires copy trait for Condition , and we cant have Copy trait for condition as it wraps serde Map which does not have the Copy trait either 

```
cannot move out of dereference of `Cac<superposition_types::Condition>`
move occurs because value has type `superposition_types::Condition`, which does not implement the `Copy` trait
```